plugin-management-cli/src/test/java/io/jenkins/tools/pluginmanager/cli/CliOptionsTest.java

```diff
@@ -0,0 +1,164 @@
+package io.jenkins.tools.pluginmanager.cli;
+
+import io.jenkins.tools.pluginmanager.config.Config;
+import io.jenkins.tools.pluginmanager.config.Settings;
+import java.io.File;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({CliOptions.class, System.class})
+public class CliOptionsTest {
+    private CliOptions options;
+    private CmdLineParser parser;
+
+    @Before
+    public void createParser() throws CmdLineException {
+        options = new CliOptions();
+        parser = new CmdLineParser(options);
+    }
+
+
+    @Test
+    public void setupDefaultsTest() throws CmdLineException {
+        parser.parseArgument();
+
+        mockStatic(System.class);
+
+        when(System.getenv("JENKINS_UC")).thenReturn("");
+        when(System.getenv("JENKINS_UC_EXPERIMENTAL")).thenReturn("");
+        when(System.getenv("JENKINS_INCREMENTALS_REPO_MIRROR")).thenReturn("");
+
+        Config cfg = options.setup();
+
+        assertEquals(Settings.DEFAULT_PLUGIN_DIR.toString(), cfg.getPluginDir().toString());
+        assertEquals(Settings.DEFAULT_WAR, cfg.getJenkinsWar());
+        assertEquals(false, cfg.isShowAllWarnings());
+        assertEquals(false, cfg.isShowWarnings());
+        assertEquals(Settings.DEFAULT_UPDATE_CENTER_LOCATION, cfg.getJenkinsUc().toString());
+        assertEquals(Settings.DEFAULT_EXPERIMENTAL_UPDATE_CENTER_LOCATION,
+                cfg.getJenkinsUcExperimental().toString());
+        assertEquals(Settings.DEFAULT_INCREMENTALS_REPO_MIRROR_LOCATION,
+                cfg.getJenkinsIncrementalsRepoMirror().toString());
+    }
+
+
+    @Test
+    public void setupAliasTest() throws CmdLineException {
+        String pluginTxtFile = this.getClass().getResource("/plugins.txt").toString();
+        String jenkinsWar = this.getClass().getResource("/jenkinstest.war").toString();
+        File file = new File(pluginTxtFile);
+        String pluginDir = file.getParent() + "/plugins";
```

Comment:
I think you want to use junit's TemporaryFolder rule here.  At the beginning of your test, you'll define the rule like so:
 ```
@Rule
    public TemporaryFolder tempFolder = new TemporaryFolder();
```
Then later you can use:
```suggestion
        String pluginDir = tempFolder.newFolder("testfolder");
```